R/utilities.R

```diff
@@ -361,60 +363,90 @@ get_ssh_session = function(host="gphost01.bcgsc.ca"){
 gene_to_region = function(gene_symbol,
                           ensembl_id,
                           genome_build = "grch37",
-                          return_as = "region"){
-
+                          return_as = "region",
+                          sort_regions = TRUE,
+                          na_for_genes_not_found = FALSE){
+  
+  stopifnot('`genome_build` parameter must be "grch37" or "grch38"' = genome_build %in% c("grch37", "grch38"))
+  stopifnot('`return_as` parameter must be "region", "bed" or "df"' = return_as %in% c("region", "bed", "df"))
+  
   #set mart based on selected genome projection
   if(genome_build == "grch37"){
     gene_coordinates = grch37_gene_coordinates
     chr_select = paste0(c(c(1:22),"X","Y"))
-  }else if(genome_build == "hg38"){
```

Comment:
This trusts the user to only ever specify one of these two options. Would be better to leave the original check for hg38 and have a trailing else here to catch the case when a user specifies an option that isn't handled.

Reply:
I think we should use either `grch37/grch38` or `hg19/hg38`. Using this mix of nomenclature like `grch37/hg38` is a bit weird for me. Please, let me know what you think.

The line 
```
stopifnot('`genome_build` parameter must be "grch37" or "grch38"' = genome_build %in% c("grch37", "grch38"))
```
makes sure that the user doesn't enter a wrong value.

Besides that, I think we always should mention what are the possible values for an argument in its documentation. Here is an example I think we should correct,
```
#' @param genome_build Reference genome build for the coordinates in the MAF file. The default is grch37 genome build.
```

Reply:
I agree sticking to the same system would make a lot of sense and simplify significantly the amount of time and effort needed to handle the chromosome prefixes, but that's not how it happened historically and most of the tools either support non-prefixed data in grch37 projection or chr-prefixed in hg38, so this convention is in the core of GAMBLR (and other projects) as well. This way, the data generated/obtained here is directly compatible with tools like vcf2maf, mutsig, oncodrive etc. So I think it is good to keep here the grch37/hg38 format for consistency and modify this stopifnot to catch the cases that aren't handled.

Reply:
Historically, our two main genome builds in GAMBL are hg38 and grch37. Accordingly, our projections have been named that way since a projection is really just a result that has been "converted" to a new coordinate system. We need to stick with hg38 and grch37 even if it's confusing. There are other nuances such as the fact that hg38 has chr-prefixed coordinates and grch37 doesn't and we try to keep things consistent with the underlying genome build associated with the projection.

Reply:
I think there is confusion regarding what I said. If I understand correctly, `grch37/grch38` are two versions of the human genome assembly and these are the nomenclature used by NCBI. `hg19/hg38` are technically the "same" versions but are the nomenclature used by UCSC.
Just to be clear, I'm not saying that we should always use the same assembly version, but the same nomenclature. 
grch37 = hg19 and grch38 = hg38.

Reply:
It's more nuanced than that though. hg19 is not the same as grch37. grch38 is not the same as hg38. Our two projections are:
grch37 (not hg19)
hg38 (not grch38)

The main differences are:
hg19, hg38 - chr-prefixed chromosome names
grch37, grch38 - bare chromosome names (do not start with chr prefix)
*the mitochondrial chromosome and other chromosomes may also have distinctions. I can't recall off the top of my head

Reply:
Yes, they are not exactly the same. That is the reason why I wrote that `grch37/grch38` and `hg19/hg38` are 
```
technically the "same" versions
```
with `same` in quotes. 

But now I understood. We really have `grch37` and `hg38` and it's not only the nomenclature. I totally agree with your change.

Reply:
Exactly. If a user requests anything other than grch37 or hg38 we should throw an error for them since we don't support any other reference versions.

Reply:
I changed the possible values for the `genome_build` argument to `grch37` or `hg38`.